src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java

```diff
@@ -0,0 +1,156 @@
+import roboy.linguistics.Linguistics;
+import roboy.linguistics.Triple;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.Neo4jRelationships;
+import roboy.memory.nodes.Interlocutor;
+import roboy.memory.nodes.Interlocutor.RelationshipAvailability;
+import static roboy.memory.nodes.Interlocutor.RelationshipAvailability.*;
+import roboy.memory.nodes.MemoryNodeModel;
+import roboy.dialog.Segue;
+import roboy.util.RandomList;
+
+import java.util.List;
+import java.util.Set;
+
+import static roboy.memory.Neo4jRelationships.*;
+
+
+/**
+ * This state will:
+ * - ask the interlocutor for his name
+ * - query memory if the person is already known
+ * - create and update the interlocutor in the context
+ * - take one of two transitions: knownPerson or newPerson
+ *
+ * IntroductionState interface:
+ * 1) Fallback is not required.
+ * 2) Outgoing transitions that have to be defined:
+ *    - knownPerson:    following state if the person is already known
+ *    - newPerson:      following state if the person is NOT known
+ * 3) No parameters are used.
+ */
+public class IntroductionState extends State {
+    private final String UPDATE_KNOWN_PERSON = "knownPerson";
+    private final String LEARN_ABOUT_PERSON = "newPerson";
+    private final Logger logger = LogManager.getLogger();
+
+    private Neo4jRelationships[] predicates = { FROM, HAS_HOBBY, WORK_FOR, STUDY_AT };
+    private State nextState;
+
+    public IntroductionState(String stateIdentifier, StateParameters params) {
+        super(stateIdentifier, params);
+    }
+
+    @Override
+    public Output act() {
+        // TODO: add more different introduction phrases
+        return Output.say("I'm Roboy. What's your name?");
+    }
+
+    @Override
+    public Output react(Interpretation input) {
+
+        // expecting something like "I'm NAME"
+
+        // 1. get name
+        String name = getNameFromInput(input);
+
+        if (name == null) {
+            // input couldn't be parsed properly
+            // TODO: do something intelligent if the parser fails
+            nextState = this;
+            logger.warn("IntroductionState couldn't get name! Staying in the same state.");
+            return Output.say("Sorry, my parser is out of service.");
+            // alternatively: Output.useFallback() or Output.sayNothing()
+        }
+
+
+        // 2. get interlocutor object from context
+        // this also should query memory and do other magic
+        Interlocutor person = Context.getInstance().ACTIVE_INTERLOCUTOR.getValue();
+        person.addName(name);
+
+
+        // 3. update interlocutor in context
+        updateInterlocutorInContext(person);
+
+        // 4. check if person is known/familiar
+        if (person.FAMILIAR) {
+
+            // 4a. person known/familiar
+            String retrievedResult = "";
+            RandomList<MemoryNodeModel> nodes = retrieveNodesFromMemoryByIds(person.getRelationships(Neo4jRelationships.FRIEND_OF));
```

Comment:
Long line, a line break wouldn't hurt. Please don't hate me for having a small monitor :)

Reply:
Refactored